install.sh

```diff
@@ -6,141 +6,142 @@ DIR="$( cd "$( dirname "$0" )" && pwd )"
+            echo "Installing as root not supported"
+            exit 1
+        fi
+        BROWSER_PATHS=( "Chrome:$HOME/.config/google-chrome/NativeMessagingHosts"
+        "Chromium:$HOME/.config/chromium/NativeMessagingHosts"
+        "Firefox:$HOME/.mozilla/native-messaging-hosts"
+        "Vivaldi:$HOME/.config/vivaldi/NativeMessagingHosts" )
+        ;;
+    *)
+        echo "$OPERATING_SYSTEM is not supported"
+        exit 1
+        ;;
 esac
 
 if [ -e "$DIR/browserpass" ]; then
-  echo "Detected development binary"
-  HOST_FILE="$DIR/browserpass"
+    echo "Detected development binary"
+    HOST_FILE="$DIR/browserpass"
 fi
 
-echo ""
-echo "Select your browser:"
-echo "===================="
-echo "1) Chrome"
-echo "2) Chromium"
-echo "3) Firefox"
-echo "4) Vivaldi"
-echo -n "1-4: "
-read BROWSER
-echo ""
-
-# Set target dir from user input
-case $BROWSER in
-1)
-  BROWSER_NAME="Chrome"
-  TARGET_DIR="$TARGET_DIR_CHROME"
-  ;;
-2)
-  BROWSER_NAME="Chromium"
-  TARGET_DIR="$TARGET_DIR_CHROMIUM"
-  ;;
-3)
-  BROWSER_NAME="Firefox"
-  TARGET_DIR="$TARGET_DIR_FIREFOX"
-  ;;
-4)
-  BROWSER_NAME="Vivaldi"
-  TARGET_DIR="$TARGET_DIR_VIVALDI"
-  ;;
-*)
-  echo "Invalid selection.  Please select 1-4."
-  exit 1
-  ;;
-esac
-
-echo "Installing $BROWSER_NAME host config"
+no_browser_found=true;
 
-# Create config dir if not existing
-mkdir -p "$TARGET_DIR"
+for browser in "${BROWSER_PATHS[@]}" ; do
+    browser_name=${browser%%:*}
+    browser_path=${browser#*:}
 
-# Escape host file
-ESCAPED_HOST_FILE=${HOST_FILE////\\/}
+    if [ -d $(dirname "$browser_path") ]; then
+        no_browser_found=false
+        read -r -p "Detected $browser_name. Install for this browser? [Y/n] " response
+        case "$response" in
+            [yY][eE][sS]|[yY])
+                install_host_config $browser_name $browser_path
+                ;;
+            *)
+                ;;
+        esac
+    fi
+done
 
-# Copy manifest host config file
-if [ "$BROWSER_NAME" == "Chrome" ] || \
-   [ "$BROWSER_NAME" == "Chromium" ] || \
-   [ "$BROWSER_NAME" == "Vivaldi" ]; then
-  if [ ! -f "$DIR/chrome-host.json" ]; then
-    echo "ERROR: '$DIR/chrome-host.json' is missing."
-    echo "If you are running './install.sh' from a release archive, please file a bug."
-    echo "If you are running './install.sh' from the source code, make sure to follow CONTRIBUTING.md on how to build first."
-    exit 1
-  fi
-  cp "$DIR/chrome-host.json" "$TARGET_DIR/$APP_NAME.json"
-else
-  if [ ! -f "$DIR/firefox-host.json" ]; then
-    echo "ERROR: '$DIR/firefox-host.json' is missing."
-    echo "If you are running './install.sh' from a release archive, please file a bug."
-    echo "If you are running './install.sh' from the source code, make sure to follow CONTRIBUTING.md on how to build first."
+if [ "$no_browser_found" = true ]; then
+    echo "No compatible browsers found."
+    echo "If  you do  actually have a  browser installed  which you  think should  be supported,"
```

Comment:
Is this a simulation of Word's "text align - justify" here? 😄

Reply:
Yep 😄 It really bothered me how the line ends were *almost* aligned, but not quite. Idk, it was a bit of an experiment. If you think it looks ugly I can still revert it 🤷‍♂️

Reply:
It looks a little bit weird to me, let's revert 🙂